### PR TITLE
Do not assume secret file location, extract a few functions.

### PR DESCRIPTION
**Pull request by vmahedia**

### CODE REVIEW

File: cli/roger_push.py [hunk 1]
```diff
@@ -385,19 +370,17 @@ def main(self, settings, appConfig, frameworkObject, hooksObj, args):
...
-                            secrets_dir, containerConfig, args, environment))
+                    if '\"SECRET\"' in output and not args.secrets_file:
+                        raise ValueError("Secrets present, need secret file to populate secrets")
+                    output = self.mergeSecrets(output, secret_vars)
                     if output != "StandardError":
                         try:
-                            comp_exists = os.path.exists("{0}".format(comp_dir))
-                            if comp_exists is False:
+                            comp_dir_exists = os.path.exists("{0}".format(comp_dir))
+                            if comp_dir_exists is False:
```
Comment (FierceElements) [inline cli/roger_push.py]: if not comp_dir_exists:

On line 382 as well. Might as well fix the sillyness as we touch the code.

File: cli/roger_push.py [hunk 2]
```diff
@@ -210,76 +241,40 @@ def main(self, settings, appConfig, frameworkObject, hooksObj, args):
...
-
+            environment = self.getTargetEnvironment(roger_env, args)
             # ----------------------------------------------
             # GetEnvironmentConfig(environment)
             # ----------------------------------------------
-            environmentObj = roger_env['environments'][environment]
-            common_repo = config.get('repo', '')
+            try:
+                environmentObj = roger_env['environments'][environment]
+            except KeyError as e:
```
Comment (FierceElements) [inline cli/roger_push.py]: No reason to include "as e" if 'e' doesn't get used.

Comment (vmahedia) [reply cli/roger_push.py]: It should be printed, made the change.

File: cli/roger_push.py [hunk 3]
```diff
@@ -210,76 +241,40 @@ def main(self, settings, appConfig, frameworkObject, hooksObj, args):
...
-                    configured_container_list.append(task.keys()[0])
-                else:
-                    configured_container_list.append(task)
-            if not set(container_list) <= set(configured_container_list):
-                raise ValueError("List of containers [{}] passed do not match list of acceptable containers: [{}]".format(
-                    container_list, configured_container_list))
+            container_list = self.getContainersList(args.app_name)
+            configured_container_list = self.getConfiguredContainersList(data)
+
+            if set(container_list) > set(configured_container_list):
```
[the remark this answers is not in the document]
Comment (vmahedia) [reply cli/roger_push.py]: s1 is neither lesser than nor equal to s2. so s1 has to be > s2

File: cli/roger_push.py [hunk 4]
```diff
@@ -210,76 +241,40 @@ def main(self, settings, appConfig, frameworkObject, hooksObj, args):
...
             framework = frameworkObj.getName()
+            common_repo = config.get('repo', '')
 
-            repo = ''
-            if common_repo != '':
-                repo = data.get('repo', common_repo)
-            else:
-                repo = data.get('repo', app_name)
-
+            repo = self.getRepository(data, common_repo, args.app_name)
```
[the remark this answers is not in the document]
Comment (vmahedia) [reply cli/roger_push.py]: Most of the function being extracted are not clean enough to be called from anywhere else. The thing is, it's very hard to read 100s of lines of code, so I am just putting the code on the side in bits and pieces. Then my plan is to look at all this functions see what concepts emerge so we can put the code in appropriate classes. 

For example, the very obvious class is an instance of a Configuration which is only single instance and the way config is operated upon at every other line is not necessary, that behavior should all be encapsulated in one class only and everyone else should be using getters. Occasionally setting some variables but they should not modify original configuration, they should be stored separately as derived configuration.

The thing is, there's a lot of improvement needed in CLI code, first we are trying to implement the features necessary for v2 so our customers stop complaining and then we can take our time to refactor and write tests suites too.

**Pull request merged**